grpc/src/test/java/com/linecorp/armeria/client/grpc/GrpcServiceImplErrorFromClientViewTest.java

```diff
@@ -0,0 +1,325 @@
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final GrpcService grpcService =
+                    GrpcService
+                            .builder()
+                            .addService(
+                                    new ServerImplErrorAtMetadataService())
+                            .build();
+            sb.service(grpcService);
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension serverWithBlockingExecutor = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final GrpcService grpcService =
+                    GrpcService.builder()
+                               .addService(
+                                       new ServerImplErrorAtMetadataService())
+                               .enableUnframedRequests(true)
+                               .supportedSerializationFormats(
+                                       GrpcSerializationFormats.values())
+                               .useBlockingTaskExecutor(true)
+                               .build();
+            sb.service(grpcService);
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+
+    // Normal case of #onError at Unary RPC.
+    @Test
+    void clientUnaryCall2() throws InterruptedException {
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            assertThatThrownBy(() -> {
+                final TestServiceBlockingStub client =
+                        GrpcClients.newClient("http://127.0.0.1:" + serverWithBlockingExecutor.httpPort(),
+                                              TestServiceBlockingStub.class)
+                                   .withDeadlineAfter(200, TimeUnit.SECONDS);
+                client.unaryCall2(REQUEST_MESSAGE);
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                    assertThat(metadata).isNotEqualTo(validMetadata);
+                });
+            });
+
+            final RequestLog log = captor.get().log().whenComplete().join();
+            assertThat(log.responseStatus()).isEqualTo(HttpStatus.OK);
+            assertThat(log.responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
+                    .satisfies(cause -> {
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                        assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                            assertThat(metadata).isNotEqualTo(validMetadata);
+                        });
+                    });
+        }
+    }
+
```

Comment:
@ikhoon 

I've created new test class from the view point of client.
This class
- uses `GrpcClients.newClient` instead of raw `ManagedChannel`.
- uses `Clients.newContextCaptor()` instead of `ServerExtension#requestContextCaptor`.

It's interesting to compare two test classes side by side.
These two classes are just to compare and confirm current behaviors as of now.
I suspect we need to tidy up response behavior about what should be set.

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/1238953/223299888-737a4b89-d56c-4135-80e2-44cb7b87fdf8.png">

I think I didn't understand what you said here properly,
https://github.com/line/armeria/pull/4686#discussion_r1124165001
but I suspect this might be related to unexpected behavior (e.g. in steaming case, error has occurred at server side, but `GrpcClients.newClient` says it has cancelled the request).
I'm very appreciated if you point out each unexpected line 🙇

Reply:
> I think I didn't understand what you said here properly,

Could you check the gRPC status code on the client side? If a response is [trailers-only](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses), a gRPC status can be acquired from `RequestLog.responseHeaders()`. Otherwise, `RequestLog.responseTrailers()` includes a gRPC status.
```java
final RequestLog log = captor.get().log().whenComplete().join();
String grpcStatus = log.responseHeaders().get(GrpcHeaderNames.GRPC_STATUS);
if (grpcStatus == null) {
    grpcStatus = log.responseTrailres().get(GrpcHeaderNames.GRPC_STATUS);
}

assertThat(grpcStatus).isNotNull();
// Maybe an internal error?
assertThat(grpcStatus).isEqualTo("13");
```

This case also need to cover the interceptor case mentioned in https://github.com/line/armeria/pull/4686#discussion_r1125403731